src/scripts/synthetic_graph_generator.py

```diff
@@ -1,15 +1,32 @@
+import argparse
+from decimal import Decimal
 import json
 import random as rand
 import re
 
 import networkx as nx
 from networkx.classes import DiGraph
+from src.weight_type import WEIGHT_TYPE,types
+
 
 
 def _get_weight(weights):
     if rand.choices([True, False], weights)[0]:
-        return round(rand.uniform(0.0, 30.0), 2)
-    return round(rand.uniform(-10.0, -1.0), 2)
+        if WEIGHT_TYPE == Decimal:
+            rand_int = rand.randint(0,30000)
+            return Decimal(rand_int) / Decimal('1000')
+        elif WEIGHT_TYPE == int:
+            return rand.randint(0,30)
+        else:
+            return round(rand.uniform(0.0, 30.0), 2)
+    if WEIGHT_TYPE == Decimal:
+        rand_int = rand.randint(-10000,-1000)
+        return Decimal(rand_int) / Decimal(1000)
+    elif WEIGHT_TYPE == int:
+        return rand.randint(-10,-1)
+    else:
+        return round(rand.uniform(-10.0, -1.0), 2)
```

Comment:
Linjen over skal også slettes, fik bare ikke lov til at gøre det, da du ikke har ændret i den. 
```suggestion
is_pos = rand.choices([True, False], weights)[0]

if WEIGHT_TYPE == Decimal:
    start, end = 0, 30_000 if is_pos else -10_000, -1000
    rand_int = rand.randint(start, end)
    return Decimal(rand_int) / Decimal(1000)

elif WEIGHT_TYPE == float:
    start, end = 0.0, 30.0 if is_pos else -10.0, -0.01 
    return round(rand.uniform(start, end), 2)

else:
    start, end = 0, 30 if is_pos else -10, -1
    return rand.randint(start, end)
```